server/controllers/profile.js

```diff
@@ -0,0 +1,123 @@
+const Profile = require("../models/profile");
+const asyncHandler = require("express-async-handler");
+const profile = require("../models/profile");
+
+// @route POST /profile/create
+// @desc create new Profile
+exports.createProfile=asyncHandler(async(req,res,next)=>{
+    const { user } = req.user.id;
+
+    const profileExists = await profile.findOne({ user });
+  
+    if (profileExists) {
+      res.status(400);
+      throw new Error("A user with that profile already exists");
+    }
+    const {
+        firstName,
+        lastName,
+        description,
+        phoneNo,
+        photoUrl,
+        email,
+        city,
+        DateOfBirth,
+        avaliabilty
+    }=req.body;
+
+  const newProfile= await profile.create({
+        firstName,
+        lastName,
+        description,
+        phoneNo,
+        photoUrl,
+        email,
+        city,
+        DateOfBirth,
+        avaliabilty,
+        user
+    })
+    if (newProfile) {
+        res.status(201).json({
+          success: {
+              id: newProfile._id,
+            }
+        });
+      } else {
+        res.status(400);
```

Comment:
```suggestion
        res.status(500);
```
A `500` seems more appropriate here